### PR TITLE
JAPI-437 TLK Indexes not filtering partitions as expected

### PR DESCRIPTION
**Pull request by jpmcmu**
- Fixed off by one error where first partition range was not created
- Added log message for invalid filters
Signed-off-by: James McMullan <James.McMullan@lexisnexis.com>

## Type of change:
- [X] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [X] I have created a corresponding JIRA ticket for this submission
- [X] My code follows the code style of this project.
  - [ ] I have applied the Eclipse code-format template provided.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [X] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [X] The change has been fully tested:
  - [ ] I have performed unit tests to cover my changes.
  - [ ] I have performed system test and covered possible regressions and side effects.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [X] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.

### CODE REVIEW

**Review by drealeed — APPROVED** [review]
LGTM

File: dfsclient/src/test/java/org/hpccsystems/dfs/client/DFSIndexTest.java [hunk 1]
```diff
@@ -109,60 +109,57 @@ public void hpccTLKFilterTest() throws Exception
...
-            HPCCFile file = new HPCCFile(indexName, connString , hpccUser, hpccPass);
 
+            HPCCFile file = new HPCCFile(indexName, connString , hpccUser, hpccPass);
             assertTrue(file.isTlkIndex());
 
             DataPartition[] fileParts = file.getFileParts();
-
-            // Last two partitions have the same range so only check up length-2
-            for (int j = 0; j < fileParts.length-2; j++)
+
```
Comment (rpastrana) [inline dfsclient/src/test/java/org/hpccsystems/dfs/client/DFSIndexTest.java:117]: very minor, remove blank spaces

File: dfsclient/src/test/java/org/hpccsystems/dfs/client/DFSIndexTest.java [hunk 2]
```diff
@@ -109,60 +109,57 @@ public void hpccTLKFilterTest() throws Exception
...
-                    Assert.fail("PartitionProcessor: " + j 
-                              + " not return expected partitions"
-                              + partitionListToString(matchedPartitions));
+                    // Due to how TLK works we can get more that one partition
+                    boolean hadExpectedPartition = false;
+                    for (int k = 0; k < matchedPartitions.size(); k++)
+                    {
+                        if (matchedPartitions.get(k).index() == j)
+                        {
+                            hadExpectedPartition = true;
```
Comment (rpastrana) [inline dfsclient/src/test/java/org/hpccsystems/dfs/client/DFSIndexTest.java:151]: can we break out of the loop at this point?

**Review by rpastrana — CHANGES_REQUESTED** [review]
@jpmcmu a couple of minor comments

**Comment by jpmcmu** [comment]
@rpastrana implemented code review changes and squashed

**Pull request merged**